File: s-recursos/src/main/java/co/edu/javeriana/arqui/rest/controller/ControladorRecursos.java
package co.edu.javeriana.arqui.rest.controller;

import co.edu.javeriana.arqui.rest.ejb.ServicioClinicas;
import co.edu.javeriana.arqui.rest.ejb.ServicioMedicamentos;
import co.edu.javeriana.arqui.rest.entity.Clinica;
import co.edu.javeriana.arqui.rest.entity.Medicamento;
import jakarta.ejb.EJB;
import jakarta.ejb.Stateless;
import java.util.List;

@Stateless
public class ControladorRecursos {

    @EJB
    private ServicioMedicamentos servicioMedicamentos;

    @EJB
    private ServicioClinicas servicioClinicas;

    public Medicamento registrarNuevoMedicamento(String nombre, String dosis, String precio) {
        Medicamento medicamento = new Medicamento();
        medicamento.setNombre(nombre);
        medicamento.setDosis(dosis);
        medicamento.setPrecio(Double.valueOf(precio));
        return servicioMedicamentos.registrarMedicamento(medicamento);
    }


    public List<Medicamento> listarMedicamentos() {
        return servicioMedicamentos.listarMedicamentos();
    }

    public Clinica registrarNuevaClinica(String nombre, String direccion, String telefono) {
        Clinica clinica = new Clinica();
        clinica.setNombre(nombre);
        clinica.setDireccion(direccion);
        clinica.setTelefono(telefono);
        return servicioClinicas.registrarClinica(clinica);
    }


    public List<Clinica> listarClinicas() {
        return servicioClinicas.listarClinicas();
    }
}

File: s-recursos/src/main/java/co/edu/javeriana/arqui/rest/controller/ControladorRecursos.kt
package co.edu.javeriana.arqui.rest.controller

import co.edu.javeriana.arqui.rest.ejb.ServicioClinicas
import co.edu.javeriana.arqui.rest.ejb.ServicioMedicamentos
import co.edu.javeriana.arqui.rest.entity.Clinica
import co.edu.javeriana.arqui.rest.entity.Medicamento
import jakarta.ejb.EJB
import jakarta.ejb.Stateless

@Stateless
open class ControladorRecursos {

    @EJB
    private lateinit var servicioMedicamentos: ServicioMedicamentos

    @EJB
    private lateinit var servicioClinicas: ServicioClinicas

    fun registrarNuevoMedicamento(nombre: String, laboratorio: String?): Medicamento {
        val medicamento = Medicamento().apply {
            this.nombre = nombre
            this.laboratorio = laboratorio
        }
        return servicioMedicamentos.registrarMedicamento(medicamento)
    }

    fun listarMedicamentos(): List<Medicamento> {
        return servicioMedicamentos.listarMedicamentos()
    }

    fun registrarNuevaClinica(nombre: String, direccion: String?): Clinica {
        val clinica = Clinica().apply {
            this.nombre = nombre
            this.direccion = direccion
        }
        return servicioClinicas.registrarClinica(clinica)
    }

    fun listarClinicas(): List<Clinica> {
        return servicioClinicas.listarClinicas()
    }
}